governance/src/components/Wallet/AccountStatus.tsx

```diff
@@ -0,0 +1,390 @@
+        return `
+          opacity: 1;
+        `;
+      case "hiding":
+        return `
+          opacity: 0;
+        `;
+      case "hidden":
+        return `
+          visibility: hidden;
+          height: 0;
+          width: 0;
+          opacity: 0;
+        `;
+    }
+  }}
+`;
+
+const MenuCloseItem = styled(MenuItem)`
+  position: absolute;
+  bottom: 50px;
+  left: 0;
+  width: 100%;
+  display: flex;
+  justify-content: center;
+`;
+
+interface AccountStatusProps {
+  variant: "desktop" | "mobile";
+}
+
+const AccountStatus: React.FC<AccountStatusProps> = ({ variant }) => {
+  const {
+    connector,
+    deactivate: deactivateWeb3,
+    library,
+    active,
+    account,
+  } = useWeb3React();
+  const [, setShowConnectModal] = useConnectWalletModal();
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+  const [copyState, setCopyState] = useState<"visible" | "hiding" | "hidden">(
+    "hidden"
+  );
+
+  // Track clicked area outside of desktop menu
+  const desktopMenuRef = useRef(null);
+  useOutsideAlerter(desktopMenuRef, () => {
+    if (variant === "desktop" && isMenuOpen) onCloseMenu();
+  });
+
+  useEffect(() => {
+    if (library && account) {
+      addConnectEvent("header", account);
+    }
+  }, [library, account]);
+
+  useEffect(() => {
+    let timer;
+
+    switch (copyState) {
+      case "visible":
+        timer = setTimeout(() => {
+          setCopyState("hiding");
+        }, 800);
+        break;
+      case "hiding":
+        timer = setTimeout(() => {
+          setCopyState("hidden");
+        }, 200);
+    }
+
+    if (timer) clearTimeout(timer);
+  }, [copyState]);
+
+  const onToggleMenu = useCallback(() => {
+    setIsMenuOpen((open) => !open);
+  }, []);
+
+  const handleButtonClick = useCallback(async () => {
+    if (active) {
+      onToggleMenu();
+      return;
+    }
+
+    setShowConnectModal(true);
+  }, [active, onToggleMenu, setShowConnectModal]);
+
+  const onCloseMenu = useCallback(() => {
+    setIsMenuOpen(false);
+  }, []);
+
+  const handleChangeWallet = useCallback(() => {
+    setShowConnectModal(true);
+    onCloseMenu();
+  }, [onCloseMenu, setShowConnectModal]);
+
+  const handleCopyAddress = useCallback(() => {
+    if (account) {
+      copyTextToClipboard(account);
+      setCopyState("visible");
+    }
+  }, [account]);
+
+  const handleOpenEtherscan = useCallback(() => {
+    if (account) {
+      window.open(`https://etherscan.io/address/${account}`);
+    }
+  }, [account]);
+
+  const handleDisconnect = useCallback(async () => {
+    if (connector instanceof WalletConnectConnector) {
+      connector.close();
+    }
+    deactivateWeb3();
+    onCloseMenu();
+  }, [deactivateWeb3, onCloseMenu, connector]);
+
+  const renderButtonContent = () =>
+    active && account ? (
+      <>
+        <Indicator connected={active} />
+        <WalletButtonText connected={active}>
+          {truncateAddress(account)} <ButtonArrow isOpen={isMenuOpen} />
```

Comment:
Needs to include the changes that [displays ENS](https://github.com/ribbon-finance/ribbon-frontend/commit/766170d3c8211890cfc8201758f7af0eefbd4882)